cns/requestcontroller/kubecontroller/crdtranslator.go

```diff
@@ -37,21 +39,22 @@ func CRDStatusToNCRequest(crdStatus nnc.NodeNetworkConfigStatus) (cns.CreateNetw
 		ncRequest.NetworkContainerid = nc.ID
 		ncRequest.NetworkContainerType = cns.Docker
 
-		// Convert "10.0.0.1/32" into "10.0.0.1" and prefix length
-		// Todo, this will be changed soon and only ipaddress will be passed
-		if ip, ipNet, err = net.ParseCIDR(nc.PrimaryIP); err != nil {
-			return ncRequest, err
+		if ip = net.ParseIP(nc.PrimaryIP); ip == nil {
+			return ncRequest, fmt.Errorf("Invalid PrimaryIP %s:", nc.PrimaryIP)
+		}
+
+		if _, size, err = GetIPAddressStringAndPrefix(nc.SubnetAddressSpace); err != nil {
```

Comment:
GetIPAddressStringAndPrefix returns a meaningful error message but you are not using it. Either log it or wrap it inside this error message (i.e. "Invalid SubnetAddressSpace ...")